src/java.base/share/classes/jdk/internal/module/ModuleReferences.java

```diff
@@ -91,8 +91,19 @@ static ModuleReference newJarModule(ModuleInfo.Attributes attrs,
                                         ModulePatcher patcher,
                                         Path file) {
         URI uri = file.toUri();
-        Supplier<ModuleReader> supplier = () -> new JarModuleReader(file, uri);
-        HashSupplier hasher = (a) -> ModuleHashes.computeHash(supplier, a);
+        String fileString = file.toString();
+        Supplier<ModuleReader> supplier = new Supplier<>() {
+            @Override
+            public ModuleReader get() {
+                return new JarModuleReader(fileString, uri);
+            }
+        };
+        HashSupplier hasher = new HashSupplier() {
+            @Override
+            public byte[] generate(String algorithm) {
+              return ModuleHashes.computeHash(supplier, algorithm);
+            }
```

Comment:
Why is JarModuleReader changed to use a file string, is this because of an environment dependency when using a Path?

Reply:
It is to avoid the following warnings during dump time:
```
[1.607s][warning][cds,heap  ] Archive heap points to a static field that may be reinitialized at runtime:
[1.607s][warning][cds,heap  ] Field: java/util/zip/ZipFile$Source::builtInFS
[1.607s][warning][cds,heap  ] Value: sun.nio.fs.LinuxFileSystem
...
[1.607s][warning][cds,heap  ] Archive heap points to a static field that may be reinitialized at runtime:
[1.607s][warning][cds,heap  ] Field: sun/nio/fs/DefaultFileSystemProvider::INSTANCE
[1.607s][warning][cds,heap  ] Value: sun.nio.fs.LinuxFileSystemProvider
```

Reply:
Thanks. At one point we will likely have to re-visit this we have prototype changes that re-implement ZipFile and java.io to use the newer APIs.